Tweak.xm

```diff
@@ -1,173 +1,129 @@
+    if (!isEnabled)
+    {
+        self.dockEdge = 4;
+        return;
     }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            CGRect origFrame = %orig;
-            CGRect fixedFrame = CGRectMake(0, 50, origFrame.size.width, origFrame.size.height);
-            return fixedFrame;
-        } else {
-            return %orig;
-        }
-    } else {
-		return %orig;
-	}
+
+    self.dockEdge = 1;
+    self.rootFolderView.dockView.hidden = shouldHide;
+    [[self.rootFolderView dockView].superview bringSubviewToFront:[self.rootFolderView dockView]];
 }
 
--(void)setFrame:(CGRect)arg1{
-    if (shouldHide) {
-        [self setHidden:TRUE];
+%end
+
+%hook SBIconListView
+
+-(id)layout
+{
+    id x = %orig;
+
+    initialized = YES;
+
+    if (!isEnabled || shouldHide)
+    {
+        self.additionalLayoutInsets = UIEdgeInsetsMake(0,0,0,0);
+        return x;
     }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            CGRect origFrame = arg1;
-            CGRect fixedFrame = CGRectMake(0, 50, origFrame.size.width, origFrame.size.height);
-            %orig(fixedFrame);
-        } else {
-            %orig;
-        }
-    } else {
-		%orig;
-	}
+
+    CGFloat dockHeight = kRootFolderController.dockHeight;
+    if ([self.iconLocation containsString:@"Root"])
+        self.additionalLayoutInsets = UIEdgeInsetsMake(dockHeight,0,-dockHeight,0);
+
+    kDockView.hidden = shouldHide;
+
+    return x;
 }
 
--(void)initWithFrame:(CGRect)arg1{
-    if (shouldHide) {
-        [self setHidden:TRUE];
-    }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            CGRect origFrame = arg1;
-            CGRect fixedFrame = CGRectMake(0, 50, origFrame.size.width, origFrame.size.height);
-            %orig(fixedFrame);
-        } else {
-            %orig;
-        }
-    } else {
-		%orig;
-	}
+%end
+
+%hook SBDockView 
+
+-(void)setHidden:(BOOL)arg 
+{
+    %orig(shouldHide ? YES : arg);
 }
 
--(CGPoint)center{
-    if (shouldHide) {
-        [self setHidden:TRUE];
-    }
-    if (@available(ios 13, *)) {
-        UIEdgeInsets safeAreaInsets = [[[UIApplication sharedApplication] keyWindow] safeAreaInsets];
-        if (safeAreaInsets.bottom != 0){
-            if (!dockPoint.x) {
-                CGPoint origPoint = %orig;
-                dockPoint = CGPointMake(origPoint.x, (origPoint.y + 50));
-            }
-            return dockPoint;
-        } else {
-            return %orig;
-        }
-    } else {
-		return %orig;
-	}
+-(BOOL)isHidden
+{
+    return shouldHide ? YES : %orig;
 }
 
 -(void)setCenter:(CGPoint)arg1{
```

Comment:
This is the only hook needed for positioning, the rest don't do anything when this is hooked.